libafl/src/bolts/ownedref.rs

```diff
@@ -166,21 +166,65 @@ where
 
 /// Wrap a slice and convert to a Vec on serialize
 #[derive(Clone, Debug)]
-pub enum OwnedSlice<'a, T: 'a + Sized> {
+enum OwnedSliceInner<'a, T: 'a + Sized> {
+    /// A ref to a raw slice and length
+    RefRaw(*const T, usize),
     /// A ref to a slice
     Ref(&'a [T]),
     /// A ref to an owned [`Vec`]
     Owned(Vec<T>),
 }
 
+/// The struct to simply wrap OwnedSliceInner to avoid exposing the pointer in it.
```

Comment:
The OwnedSliceInner is implementation detail and we probably don't want to document it.
I'd just use the original `/// Wrap a slice and convert to a Vec on serialize` comment